### DISCUSSION TITLE
MSI installation options

### DISCUSSION THREAD

**Discussion by ghost** [d2phap/ImageGlass#2049] [Q&A] [answered]
We are planning to install ImageGlass as replacement for Windows.Photos.App. Now it supports MSI installation options. But after installing and starting the program, it will ask 3 questions (Language, Pro or default User and make it the "Default App"). Is there a possibility to give those 3 option in the msi installation string? Or is it necessary to app.config-file and copy that one after installation? 

We want the less possible interference with the user. So it had to be run-ready after installation. I certainly will get some phone calls even at the first question about the Language... Yes i know my fools around here....

**Comment by d2phap** [comment]
Hi @RomanOswald,

You can create `igconfig.default.json` file in the installed folder of ImageGlass with this content:
```json
{
  "_Metadata": {
    "Description": "ImageGlass configuration file",
    "Version": "9.1"
  },
  "QuickSetupVersion": 9
}
```

More details about configuration files can be found here: https://imageglass.org/docs/app-configs

**Reply by ghost** [reply]
Thanks, that worked for a part.

It now starts directly. But there are two things to be done. First: The language is wrong, default is English, we need German. So is it enough to add   **"Language": "German.iglang.json",** somewhere?

And it is not the default app to open pictures. How can that be done?

Could not find that information on the site.

**Reply by d2phap** [reply, marked as answer]
yes, you can add `"Language": "German.iglang.json"` to the `igconfig.default.json`.

About the default app, some people create [a XML file and set it via GPO](https://groups.google.com/g/imageglass/c/-bkIUt_oJZo/m/KaSUdvAZAQAJ). You can also use ImageGlass command:

```
set-default-viewer [string exts] [--per-machine]
```

Example:
```
igcmd.exe set-default-viewer .png;.jpg;.webp --per-machine
```

https://imageglass.org/docs/command-line-utilities#set-default-viewer-string-exts-per-machine

**Reply by ghost** [reply]
Thanks, that worked. But I need to make a GPO, the set-default won't work on a script.

/Edit: So instead of using the commando igcmd it is easier to export the Reg HKCU>Software>ImageGlass and import it on the new machine, but it still does not work... The GPO also does not work... I hate Windows 11...

The only way working, is opening the Program, entering the Properties and use the "set as default button". And after that open a file and set it to default per file type. Did I mention that I hate Windows 11...

I just make a how to and put that on the intranet....
